src/code/z_actor.c

```diff
@@ -213,9 +213,9 @@ void Actor_SetFeetPos(Actor* actor, s32 limbIndex, s32 leftFootIndex, Vec3f* lef
     }
 }
 
-void func_8002BE04(GlobalContext* globalCtx, Vec3f* arg1, Vec3f* arg2, f32* arg3) {
-    SkinMatrix_Vec3fMtxFMultXYZW(&globalCtx->viewProjectionMtxF, arg1, arg2, arg3);
-    *arg3 = (*arg3 < 1.0f) ? 1.0f : (1.0f / *arg3);
+void Actor_ProjectPos(GlobalContext* globalCtx, Vec3f* src, Vec3f* xyzDest, f32* wDest) {
+    SkinMatrix_Vec3fMtxFMultXYZW(&globalCtx->viewProjectionMtxF, src, xyzDest, wDest);
+    *wDest = (*wDest < 1.0f) ? 1.0f : (1.0f / *wDest);
```

Comment:
It should be properly documented that the last argument is *not* a "w"
It's not even an inverted w since it caps it at 1

I'm personally not familiar enough with [homogeneous coordinates](https://en.wikipedia.org/wiki/Homogeneous_coordinates) to explain the part where they check ` w < 1`.
I would understand a `IS_ZERO`-like check, to avoid dividing by almost-0 and eventually making garbage float operations, but a threshold of 1 looks large so that must not be the intent

so I would leave `wDest` unnamed for now (and probably forever)

Reply:
While I don't quite understand the capping, it's obviously an inverted w like you said. It's used like 1/w to multiply the x and y coordinates to get the screen coordinates.
It should really be named thus.

Reply:
also is ProjectPos a good name for this? that sounds very generic like its just projecting the pos onto something. when this is specifically relevant to the `viewProjectionMtx`
not smart enough with view things and matrix stuff to know personally, so its a genuine question

Reply:
`viewProjectionMtx` is also used to compute `Actor#projectedPos`:
https://github.com/zeldaret/oot/blob/master/src/code/z_actor.c#L2364
```c
SkinMatrix_Vec3fMtxFMultXYZW(&globalCtx->viewProjectionMtxF, &actor->world.pos, &actor->projectedPos,
                             &actor->projectedW);
```
so yeah I think the name is good

------

about naming wDest I really wouldn't just dismiss the cap on the grounds of not knowing its purpose lol

Reply:
How about cappedInvWDest? It does encapsulate all the issues we have with it? Same for its uses in the code.